Fix asset downloading authentication error

akm opened the pull request:
Fix the following error.
It seems that `client` of `octokit` sends request with authentication header for github, but S3 which might store asset files doesn't accept the authentication header for github.
This PR changes to send the requests to S3 with `curl` command instead of `client` of `octokit` .

## Error

```
$ exe/github_export assets_download -n 1 -d tmp/msa_apps-20180409 -t xxxxxxxxxxxxxx -V
(snip)
Traceback (most recent call last):
	23: from exe/github_export:4:in `<main>'
	22: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	21: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	20: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	19: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	18: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/github_export-0.1.1/lib/github_export/command.rb:103:in `assets_download'
	17: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/github_export-0.1.1/lib/github_export/command.rb:103:in `each'
	16: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/github_export-0.1.1/lib/github_export/command.rb:104:in `block in assets_download'
	15: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/github_export-0.1.1/lib/github_export/command.rb:104:in `fork'
	14: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/github_export-0.1.1/lib/github_export/command.rb:105:in `block (2 levels) in assets_download'
	13: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/github_export-0.1.1/lib/github_export/command.rb:105:in `each'
	12: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/github_export-0.1.1/lib/github_export/command.rb:111:in `block (3 levels) in assets_download'
	11: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/octokit-4.8.0/lib/octokit/connection.rb:19:in `get'
	10: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/octokit-4.8.0/lib/octokit/connection.rb:156:in `request'
	 9: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/sawyer-0.8.1/lib/sawyer/agent.rb:94:in `call'
	 8: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/faraday-0.9.1/lib/faraday/connection.rb:140:in `get'
	 7: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/faraday-0.9.1/lib/faraday/connection.rb:377:in `run_request'
	 6: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/faraday-0.9.1/lib/faraday/rack_builder.rb:139:in `build_response'
	 5: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/octokit-4.8.0/lib/octokit/middleware/follow_redirects.rb:61:in `call'
	 4: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/octokit-4.8.0/lib/octokit/middleware/follow_redirects.rb:73:in `perform_with_redirection'
	 3: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/faraday-0.9.1/lib/faraday/response.rb:8:in `call'
	 2: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/faraday-0.9.1/lib/faraday/response.rb:57:in `on_complete'
	 1: from $HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/faraday-0.9.1/lib/faraday/response.rb:9:in `block in call'
$HOME/.anyenv/envs/rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/octokit-4.8.0/lib/octokit/response/raise_error.rb:16:in `on_complete': GET https://cloud.githubusercontent.com/assets/xxxxxxxxxxxxxxxxxxxx/xxxxxxxxxxxxxxxxxxx.png: 400 - <?xml version="1.0" encoding="UTF-8"?> (Octokit::BadRequest)
<Error><Code>InvalidArgument</Code><Message>Unsupported Authorization Type</Message><ArgumentName>Authorization</ArgumentName><ArgumentValue>token xxxxxxxxxxxxxx</ArgumentValue><RequestId>xxxxxxxxxxxxxxxxxxxxxx</RequestId><HostId>xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx</HostId></Error>
```

akm committed:
:fix: Use curl instead of octkit client to download files

akm committed:
:+1: Bmp up to 0.1.2

minimum2scp commented on lib/github_export/command.rb:
```diff
@@ -108,7 +108,12 @@ def assets_download
-              File.binwrite(dest, client.get(url))
+              # Assets might be downloaded from S3 so we use curl (or httpclient?) without auth info instead of `client` object
+              # File.binwrite(dest, client.get(url))
+              cmd = "curl -o #{dest} #{url}"
```
To detect error, [-f](https://curl.haxx.se/docs/manpage.html#-f) option is required.

without `-f` option:

```console
yamada@WEZEN% curl -o a.png https://cloud.githubusercontent.com/assets/DUMMY.png
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   243    0   243    0     0    199      0 --:--:--  0:00:01 --:--:--   199
yamada@WEZEN% echo $?
0
yamada@WEZEN% ls -l a.png
-rw-r--r-- 1 yamada yamada 243  4月  9 14:47 a.png
yamada@WEZEN% cat a.png
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>AccessDenied</Code><Message>Access Denied</Message><RequestId>7921781BFBD86F36</RequestId><HostId>rs5HYpBjpCkQTMG7NG0pcY5GBuXmqrpDiLp4XacaBczlZgF50h7k9cvUD9LBGA7FZQRz6fqjm4o=</HostId></Error>
```

with `-f` option:

```console
yamada@WEZEN% curl -f -o a.png https://cloud.githubusercontent.com/assets/DUMMY.png
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
curl: (22) The requested URL returned error: 403 Forbidden
yamada@WEZEN% echo $?
22
```

akm replied:
@minimum2scp Thanks! I added it.

akm committed:
:+1: Add -f option to detect errors  @minimum2scp pointed at

https://github.com/groovenauts/github_export/pull/4/files/51346c276ac5a8b73e82cd48d4ffbe9a3444a6f6#diff-28722ad099bc3d1ed63a36edceb787d0
